language/move-prover/bytecode/src/read_write_set_analysis.rs

```diff
@@ -576,21 +633,23 @@ impl<'a> TransferFunctions for ReadWriteSetAnalysis<'a> {
                         .bind_local(*lhs, AbsAddr::constant(a.clone()), func_env)
                 }
             }
-            Assign(_attr_id, lhs, rhs, _assign_kind) => {
-                if let Some(rhs_data) = state.locals.get_local(*rhs, func_env).cloned() {
-                    state.locals.bind_local(*lhs, rhs_data, func_env)
-                } else {
-                    state.locals.remove_local(*lhs, func_env)
-                }
-            }
+            Assign(_attr_id, lhs, rhs, _assign_kind) => state.assign_local(*lhs, *rhs, func_env),
```

Comment:
I think this changes the behaviour compared to the previous version as `assign_local` seems to ignore cases where it cannot find data for rhs but I'm not sure which behaviour is the desired one.
What would it mean in terms of code for an assignment's rhs not to have data?

Reply:
> I think this changes the behaviour compared to the previous version as assign_local seems to ignore cases where it cannot find data for rhs but I'm not sure which behaviour is the desired one.

Good call--this does change the behavior in two (intended) ways:
- Previously, if the RHS was an abstraction of a struct, it would not get bound to the RHS
- Previously, if the RHS was not found, the LHS would be removed. I don't think this code was ever exercised

> What would it mean in terms of code for an assignment's rhs not to have data?

This is related to the discussion above about only abstracting addresses and structs--the rhs will have no data if it is (e.g.) and integer or boolean type.

Reply:
Got it, thanks for the explanation!